src/v2/Apps/Fair/Routes/FairExhibitors.tsx

```diff
@@ -1,46 +1,120 @@
-import React from "react"
-import { createFragmentContainer, graphql } from "react-relay"
+import { Button, Col, Grid, Row } from "@artsy/palette"
+import React, { useState } from "react"
+import {
+  RelayPaginationProp,
+  createPaginationContainer,
+  graphql,
+} from "react-relay"
 import { FairExhibitors_fair } from "v2/__generated__/FairExhibitors_fair.graphql"
 import { FairExhibitorRailFragmentContainer as FairExhibitorRail } from "../Components/FairExhibitorRail"
 
+enum Mode {
+  Resting,
+  Loading,
+  Error,
+}
+
 interface FairExhibitorsProps {
   fair: FairExhibitors_fair
+  relay: RelayPaginationProp
 }
 
-const FairExhibitors: React.FC<FairExhibitorsProps> = ({ fair }) => {
+const FairExhibitors: React.FC<FairExhibitorsProps> = ({ fair, relay }) => {
+  const [mode, setMode] = useState(Mode.Resting)
+
+  const handleClick = () => {
+    if (!relay.hasMore() || relay.isLoading()) return
+
+    setMode(Mode.Loading)
+
+    relay.loadMore(15, err => {
+      if (err) {
+        setMode(Mode.Error)
```

Comment:
Do we use these `Error` or `Resting` modes at all?

Reply:
`Resting` yes since it's the default. `Error` — not explicitly other than it switches it out of `Loading`. There isn't really an error state defined, like typically I'd imagine maybe changing the copy on the button, and disabling it, then timing out a reset to `Resting` after a second or two. Currently if an error happens it just silently resets the button and nothing else.

(I generally think using boolean flags to represent a components overall state a bad idea but if `isLoading` is clearer for others I'm open to it.)

Reply:
Since we don't have an error state yet (although the UX you describe sounds nice 😍 ), a simple flag might be easier to grok.